week3/homework/src/index.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+const express = require('express');
+const app = express();
+const bodyParser = require('body-parser');
+
+const { readTodo, clearTodos, markAsDone, markAsNotDone } = require('./actions');
+
+const Todo = require('./todo');
+const FILENAME = `${__dirname}/todoList.json`;
+const todo = new Todo(FILENAME);
+
+app.use(bodyParser.json());
+
+const TODO_SLUG = 'todos';
+// Get a single to-do with ID :id
+app.get(`/${TODO_SLUG}/:id`, readTodo.bind(null, todo));
+
+// Clear the list of to-dos
+app.delete(`/${TODO_SLUG}`, clearTodos.bind(null, todo));
+
+// Set the done flag of a single to-do to true
+app.post(`/${TODO_SLUG}/:id/done`, markAsDone.bind(null, todo));
+
+// Set the done flag of a single to-do to false
+app.delete(`/${TODO_SLUG}/:id/done`, markAsNotDone.bind(null, todo));
```

Comment:
This one is also the same as the previous one. In the homework as ```markAsNotDone (DELETE /todos/:id/done)```

Reply:
Good question! And you are right, we can't make them both the same route and method, so we should change the route for `markAsNotDone` to  `/todos/:id/undone`

See https://restfulapi.net/http-methods/#put when to use PUT

But since you are requested to use DELETE (https://github.com/HackYourFuture/Node.js/blob/master/week3/homework/README.md#markasnotdone-delete-todosiddone); you can leave it as is.

Reply:
Thanks Samir for your reply and the link... You are right also for using PUT which is used when updating a resource. 
@eidosam